### PR TITLE
Lua update discord

### PR DESCRIPTION
**Pull request by MjnMixael**
Allow Lua to set the Discord presence to custom strings. This is intended for SCPUI where game state changes sometimes skip over the built-in logic and custom game states might warrant updating the presence as well.

### CODE REVIEW

**Review by The-E — APPROVED** [review]

File: code/libs/discord/discord.cpp [hunk 1]
```diff
@@ -272,5 +272,15 @@ void shutdown()
 	shutdown_discord();
 }
 
+void set_presence_string(SCP_string text)
```
Comment (Goober5000) [inline code/libs/discord/discord.cpp:275]: This ought to be `const SCP_string&`, like `set_presence`

**Review by Goober5000 — CHANGES_REQUESTED** [review]
Small change

File: code/libs/discord/discord.cpp (hunk 1 above)
Comment (MjnMixael) [reply code/libs/discord/discord.cpp]: done

**Review by Goober5000 — APPROVED** [review]

File: code/libs/discord/discord.h [hunk 2]
```diff
@@ -5,5 +5,7 @@ namespace discord {
 
 void init();
 void shutdown();
+void set_presence_string(SCP_string text);
```
Comment (Goober5000) [inline code/libs/discord/discord.h]: This will need `const SCP_string &` also

**Review by Goober5000 — DISMISSED** [review]

**Pull request merged**